interop-testing/src/main/java/io/grpc/testing/integration/TrafficControlProxy.java

```diff
@@ -0,0 +1,256 @@
+package io.grpc.testing.integration;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.Preconditions;
+
+import io.netty.channel.socket.nio.NioSocketChannel;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+// import java.lang.annotation.Native;
+import java.net.InetSocketAddress;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.net.SocketException;
+import java.net.SocketImpl;
+import java.util.ArrayList;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+
+public final class TrafficControlProxy {
+
+  private static int DEFAULT_BAND_BPS = 1 * 1024 * 1024;
+  private static int DEFAULT_DELAY_NANOS = 100 * 1000 * 1000;
+  private static Logger logger = Logger.getLogger(TrafficControlProxy.class.getName());
+
+  // TODO: make ports arguments (and probably loopback/ip as well)
+  private String loopback = "127.0.0.1";
+  private int proxyPort = 5050;
+  private int serverPort = 5001;
+  private int queueLength;
+  private int chunkSize;
+  private int bandwidth;
+  private long latency;
+  private final LinkedBlockingQueue<Message> serverQueue = new LinkedBlockingQueue<Message>();
+  private final LinkedBlockingQueue<Message> clientQueue = new LinkedBlockingQueue<Message>();
+  private volatile int serverQueueSize;
+  private volatile int clientQueueSize;
+  private volatile boolean shutDown;
+  private ServerSocket clientAcceptor;
+  private Socket serverSock;
+  private Socket clientSock;
+
+  /**
+   * Returns a new TrafficControlProxy with default bandwidth and latency
+   */
+  public TrafficControlProxy() {
+    this(DEFAULT_BAND_BPS, DEFAULT_DELAY_NANOS, TimeUnit.NANOSECONDS);
+  }
+
+  /**
+   * Returns a new TrafficControlProxy with bandwidth set to targetBPS, and latency set to
+   * targetLatency in latencyUnits.
+   *
+   * @param targetBPS
+   * @param targetLatency
+   * @param latencyUnits
+   */
+  public TrafficControlProxy(int targetBPS, int targetLatency, TimeUnit latencyUnits) {
+    Preconditions.checkArgument(targetBPS > 0);
```

Comment:
FYI you can statically import checkArgument.   In Eclipse you can select it and pres Ctrl Shift M.